make/autoconf/basic.m4

```diff
@@ -96,8 +96,14 @@ AC_DEFUN_ONCE([BASIC_SETUP_PATHS],
   AUTOCONF_DIR=$TOPDIR/make/autoconf
 
   # Setup username (for use in adhoc version strings etc)
-  # Outer [ ] to quote m4.
-  [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
+  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
```

Comment:
This variable is currently initiated in a rather weird location given what it's used for. As it used to be a one liner, it didn't seem worth the effort to move it to a more suitable location, but when expanding it with a --with flag I think we should also move this to jdk-version.m4, which is the only place where it's currently used. I think it makes sense to treat this as part of the version variables as that's what it's used for.

Reply:
it's also used in make/hotspot/lib/CompileJvm.gmk, but agree moving to jdk-versions.m4 makes sense

Reply:
Right, I meant the only place it's used within configure.

Reply:
done, moved